src/metadata.rs

```diff
@@ -50,13 +46,52 @@ impl From<io::Error> for MetadataError {
     }
 }
 
-fn get_date_time_created(path: &Path) -> Result<chrono::DateTime<FixedOffset>, MetadataError> {
-    Ok(FixedOffset::west(0).timestamp(fs::metadata(path)?.ctime(), 0))
+fn result_to_option<T, E: Error>(res: Result<T, E>, checkpoint: String) -> Option<T> {
+    match res {
+        Ok(t) => Some(t),
+        Err(e) => {
+            eprint!("Error occurred during {}, {}", checkpoint, e);
+            None
+        }
+    }
 }
 
-fn convert_exif_date_time_to_chrono_date_time_fixed_offset(
+fn open_file(path: &Path, file_name: &String) -> Option<fs::File> {
+    result_to_option(fs::File::open(path), "opening file ".to_owned() + file_name)
```

Comment:
Apparently a more common way to format strings is  [format!()](https://doc.rust-lang.org/std/macro.format.html)